homeassistant/components/wled/config_flow.py

```diff
@@ -115,3 +127,32 @@ def _show_confirm_dialog(self, errors: dict | None = None) -> FlowResult:
             description_placeholders={"name": name},
             errors=errors or {},
         )
+
+
+class WLEDOptionsFlowHandler(OptionsFlow):
+    """Handle WLED options."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize WLED options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage WLED options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_KEEP_MASTER_LIGHT,
```

Comment:
Is it not possible to use the entity registry disable feature instead of an options flow and sync a stored (hass.data) boolean, possibly on a dataclass, when adding/removing the master light entity?

Reply:
Yes/No.

I have considered that option (and originally tried implementing it as well when the segment support was added).

Context:

The problem is that having segments without a master light can be uncontrollable (you can't turn it on if the whole strip is off). So, it is needed when having multiple segments. Disabling the entity in those cases makes no sense.

The problem with the entity registry then becomes, should the integration control that? If the integration starts toggling the registry I think the behavior becomes weird as well. When does it turn on? When does it turn off?

The entity registry kinda assumes a default value, while the default value is unknown or changes. It can turn on/off by integration, but once the user has toggled that, it sticks as it became user-controlled and no way to release that (disabled by user).

I do agree this is a weird feature of WLED. I wanted to make sure the default behavior is as expected:

- Users with a single segment, have no master light. Just a single light entity that controls their LED strip as expected. Simple, functional and clean.
- Users with multiple segments: Light for each segment + master light for controlling the whole device.
- Users that configure their strip from a single segment -> multiple segments (or the other way around): HA will adjust.

User automating led strip segmentation is IMHO an edge case, hence I dropped it in an option instead of possible conflicting controls in the entity registry.

Reply:
Yeah, I didn't notice we create the master entity in two places possibly.